File: crates/psychevo-runtime/tests/fixtures/hook_worker_call_adapter.py
import json, sys
for line in sys.stdin:
    req=json.loads(line)
    method=req.get("method")
    if method=="initialize":
        result={"ok": True}
    elif method=="hooks/call":
        result={"feedback":"worker saw hook"}
    elif method=="shutdown":
        result={"ok": True}
    else:
        result={}
    print(json.dumps({"jsonrpc":"2.0","id":req.get("id"),"result":result}), flush=True)

File: crates/psychevo/tests/fixtures/hook_worker_call_adapter.js
#!/usr/bin/env node
const readline = require("readline");

readline.createInterface({ input: process.stdin, crlfDelay: Infinity }).on("line", (line) => {
  const request = JSON.parse(line);
  let result = {};
  if (request.method === "initialize" || request.method === "shutdown") {
    result = { ok: true };
  } else if (request.method === "hooks/call") {
    result = { feedback: "worker saw hook" };
  }
  process.stdout.write(`${JSON.stringify({ jsonrpc: "2.0", id: request.id, result })}\n`);
});
